src/asl.ts

```diff
@@ -723,6 +724,65 @@ export class ASL {
           },
         };
       });
+    } else if (isForStmt(stmt)) {
+      const body = this.evalStmt(stmt.body);
+
+      if (!body) {
+        // if the body is empty, the for expression is evaluated,
+        // and then continue on. This state will be optimized out later.
+        return {
+          Type: "Pass",
+        };
+      }
+
+      return this.evalContextToSubState(stmt, (evalExpr) => {
+        const initializer = stmt.variableDecl
+          ? isVariableStmt(stmt.variableDecl)
+            ? this.evalStmt(stmt.variableDecl)
+            : evalExpr(stmt.variableDecl)
+          : undefined;
```

Comment:
What does it mean for `variableDecl` to not be a `VariableStmt`?

Reply:
```ts
let i;
for(i in [1,23]) {}
return i;
```

Reply:
In the TS AST, the initializer is a `VariableDeclList` or `Expr`, but in the case of for in and for of, it can only be (I think?) `[VariableDecl]` or `Identifier`.

I don't like that they don't restrain the types.